### PR TITLE
Add another git submodule for C# called c-sharp-pro

### PR DESCRIPTION
**Pull request by mjambon**
This uses `tree_sitter_c_sharp_pro_` prefix in the C functions to avoid conflicts with the other C# parser so they can be linked into the same executable.

### Checklist

- [x] Any new parsing code was already published, integrated, and merged into Semgrep. DO NOT MERGE THIS PR BEFORE THE SEMGREP INTEGRATION WORK WAS COMPLETED.
- [x] Change has no security implications (otherwise, ping the security team)

### CODE REVIEW

File: lang/c-sharp-pro/ConvexHull.cs [hunk 1]
```diff
@@ -0,0 +1,322 @@
+/*
```
Comment (nmote) [inline lang/c-sharp-pro/ConvexHull.cs:1]: What's this source file doing here? Should it be in a test directory?

File: lang/language-variants-0.20.6 [hunk 2]
```diff
@@ -1,6 +1,5 @@
 apex
 bash
-c-sharp
```
Comment (nmote) [inline lang/language-variants-0.20.6:3]: Is this related to adding the new submodule?

File: lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
+/*
+ * semgrep-csharp
+ *
+ * Extend the original tree-sitter C# grammar with semgrep-specific constructs
+ * used to represent semgrep patterns.
+ *
+ * The language is renamed from c-sharp to csharp to match language name
```
Comment (nmote) [inline lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js:7]: Looks like it's actually named `c_sharp_pro` -- is this a copy/paste error?

**Review by nmote — APPROVED** [review]

File: lang/c-sharp-pro/ConvexHull.cs (hunk 1 above)
Comment (mjambon) [reply lang/c-sharp-pro/ConvexHull.cs:1]: You're right. I copied it over from the original c-sharp folder without noticing it shouldn't be here.

File: lang/language-variants-0.20.6 (hunk 2 above)
Comment (mjambon) [reply lang/language-variants-0.20.6:3]: No. I made a mistake last week when building this list. It's because the c-sharp was failing to build in CI but it was due to a lack of memory, not an incompatibility with the latest tree-sitter.

File: lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js (hunk 3 above)
Comment (mjambon) [reply lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js:7]: yes

File: lang/c-sharp-pro/ConvexHull.cs (hunk 1 above)
Comment (mjambon) [reply lang/c-sharp-pro/ConvexHull.cs:1]: um, it was in the correct place originally. I just did a wrong copy-paste.

Comment (mjambon) [reply lang/c-sharp-pro/ConvexHull.cs]: both c-sharp and c-sharp-pro now contain the same files:
```
~/ocaml-tree-sitter-semgrep/lang/c-sharp-pro $ git ls-files
Makefile
extensions.txt
fyi.list
projects.txt
test/ok/ConvexHull.cs
```

File: lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
+/*
+ * semgrep-c-sharp-pro
+ *
+ * Extend the original tree-sitter C# grammar with semgrep-specific constructs
+ * used to represent semgrep patterns.
+ *
+ * c-sharp-pro is the same as c-sharp but tracks a more recent version
```
Comment (aryx) [inline lang/semgrep-grammars/src/semgrep-c-sharp-pro/grammar.js:7]: it's a bit annoying that we have to duplicate grammar.js ... but maybe in the longer term it's better.

**Comment by aryx** [comment]
I thought it would have been simpler to use just one tree-sitter-chsarp, and update it here to the latest but update to a more
recent semgrep-c-sharp in pro, but maybe you're right it's simpler to separate them also here.

**Comment by aryx** [comment]
especially if you managed to avoid naming conflicts in C.

**Comment by aryx** [comment]
cc @yosefAlsuhaibani who might have to do something similar for another language (I don't remember which one).
